gammapy/modeling/models/prior.py

```diff
@@ -32,7 +32,15 @@ def _build_priorparameters_from_dict(data, default_parameters):
 class Prior(ModelBase):
     _unit = ""
 
-    def __init__(self, **kwargs):
+    def __init__(self, modelparameters, **kwargs):
```

Comment:
I think taking the `Parameter` object here on init is rather confusing to establish the linking between parameters and their priors. I think this also deviates from the API proposed in the PIG.